protocols/rendezvous/src/client.rs

```diff
@@ -281,6 +281,7 @@ impl NetworkBehaviour for Behaviour {
 
                     return Poll::Ready(new_to_swarm);
                 }
+                Poll::Ready(_) => {}
                 Poll::Pending => {}
```

Comment:
Just noticing this here. Continuing the loop here is a bug, we should return `Poll::Pending` from the function otherwise this function probably spins forever? Weird that we didn't notice this earlier.

Even better, we should probably use the `futures::ready!` macro.

Reply:
Would you be up for sending a separate PR for this?

Reply:
Sure I can do that and fix this. I'm not sure what the `futures::read!` is. I'll have to look it up.

Reply:
> Sure I can do that and fix this. I'm not sure what the `futures::read!` is. I'll have to look it up.

It is a macro for short-circuiting a call to a function that returns `Poll` and cascades up the `Poll::Pending` variant, a bit like `?` but for futures.

Reply:
Do you have any example code that uses it?

Reply:
Actually, nevermind. See line 297 in this diff? That is where we return `Poll::Pending`. GitHub hid that part of the diff so I didn't see it when making the comment :)